cabal-install/bootstrap.sh

```diff
@@ -52,24 +55,32 @@ SCOPE_OF_INSTALLATION="${SCOPE_OF_INSTALLATION:---user}"
 DEFAULT_PREFIX="${HOME}/.cabal"
 
 # Try to respect $TMPDIR.
-[ -"$TMPDIR"- = -""- ] &&
-  export TMPDIR=/tmp/cabal-$(echo $(od -XN4 -An /dev/random)) && mkdir $TMPDIR
+if [ -"$TMPDIR"- = -""- ]; then
+    TMPDIR=/tmp/cabal-$(od -XN4 -An /dev/random)
+    mkdir "$TMPDIR"
+    export TMPDIR
+fi
```

Comment:
@tuncer I'm actually not sure myself where it's used, but your change looks fine to me.